### ISSUE TITLE
Openssl 1.1.0b VC-WIN64-common looks for padlock.dll instead of padlock lib

### ISSUE THREAD

**Issue by Gariben** [openssl/openssl#1809]
When running

`perl Configure debug-VC-WIN64-common`

from OpenSSL 1.1.0b,
you get a number of unresolved external symbols for e_padlock.obj:

```
   Creating library engines\ossltest.lib and object engines\ossltest.exp
        IF EXIST engines\ossltest.dll.manifest  mt -nologo -manifest engines\oss
ltest.dll.manifest -outputresource:engines\ossltest.dll
        IF EXIST engines\padlock.dll.manifest DEL /F /Q engines\padlock.dll.mani
fest
        link /nologo /debug /dll /out:engines\padlock.dll /def:C:\Users\jleibeck
\AppData\Local\Temp\nm89FF.tmp @C:\Users\jleibeck\AppData\Local\Temp\nm8A00.tmp
   Creating library engines\padlock.lib and object engines\padlock.exp
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_capability re
ferenced in function _padlock_available
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_key_bswap ref
erenced in function _padlock_aes_init_key
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_reload_key re
ferenced in function _padlock_aes_init_key
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_aes_block ref
erenced in function _padlock_cfb_cipher
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_ecb_encrypt r
eferenced in function _padlock_ecb_cipher
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_cbc_encrypt r
eferenced in function _padlock_cbc_cipher
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_cfb_encrypt r
eferenced in function _padlock_cfb_cipher
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_ofb_encrypt r
eferenced in function _padlock_ofb_cipher
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_ctr32_encrypt
 referenced in function _padlock_ctr32_encrypt_glue
e_padlock.obj : error LNK2019: unresolved external symbol _padlock_xstore refere
nced in function _padlock_rand_bytes
engines\padlock.dll : fatal error LNK1120: 10 unresolved externals
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 14.0
\VC\BIN\link.EXE"' : return code '0x460'
Stop.
```

I noticed that this command references engines/padlock**.dll**, which does not exist. However, if I change all of these references in the generated makefile to padlock**.lib**, then it works. I tried to find the root of the problem but wasn't able to. Just thought I'd let you guys know.

**Comment by levitte** [comment]
Not sure what what you actually changed...  however, `/out:engines\padlock.dll` means that it's trying to *produce* `padlock.dll`.  However, that seems to fail, so that's why you can't find it.

The reason this happens is that one of `engine\e_padlock-x86.asm` or `e_padlock-x86_64.asm` weren't generated and/or compiled properly...  or so I can guess.  I have no clue why for the moment...

**Comment by Gariben** [comment]
I corrected my initial post to be the for the **debug** configuration, if that changes anything.

However it occurs in VC-WIN64-common for me as well.

The lines I change in the makefile are the following:

```
<INSTALL_ENGINES=engines\capi.dll engines\padlock.dll
>INSTALL_ENGINES=engines\capi.dll engines\padlock.lib
```

along with every other subsequent mention in the makefile, which amounts to 10 occurences total.


From what I can tell all it does it change out the padlock.dll for a padlock.lib in the output.
I'm going to check out why it isn't being generated. I'm under pressure to get this built on a variety of platforms, so if the lib works for me I may just have to move on.

**Comment by levitte** [comment]
Oh!  Now I see what you're doing!  Sorry, `VC-WIN64-common` is just a template, choose `VC-WIN64A` instead.

However, I understand why you chose this, as `VC-WIN64-common` isn't properly marked as a template.  Will fix.

**Comment by levitte** [comment]
Fix in #1811

**Comment by levitte** [comment]
Fix is now merged.  Closing this.  Thank you for the report!
